compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/CollectionStubMethodLowering.kt

```diff
@@ -136,12 +136,20 @@ private class CollectionStubMethodLowering(val context: JvmBackendContext) : Cla
         return mutableClass.typeParameters.map { it.symbol }.zip(readOnlyClassTypeArguments).toMap()
     }
 
+    // Preserve old backend's logic to generate stubs for special cases where a mutable method
+    // have same jvm signature with an immutable method.
+    private val specialNames = setOf("listIterator", "subList")
+
     private inner class StubsForCollectionClass(
         val readOnlyClass: IrClassSymbol,
         val mutableClass: IrClassSymbol
     ) {
         val mutableOnlyMethods: Collection<IrSimpleFunction> by lazy {
-            val readOnlyMethodSignatures = readOnlyClass.functions.map { it.owner.toSignature() }.toHashSet()
+            val readOnlyMethodSignatures = readOnlyClass
+                .functions
+                .filter { it.owner.name.asString() !in specialNames }
```

Comment:
Please use the full JVM signature (after the subsequent `.map { it.owner.toSignature() }` call) to filter out these members, instead of only name